main_test.go

```diff
@@ -474,4 +474,149 @@ func TestMain(t *testing.T) {
 			assert.Equal(t, chartWrappers[3].Metadata.Version, "1.0.2")
 		})
 	})
+
+	t.Run("writeCharts", func(t *testing.T) {
+		t.Run("should add charts that do not exist on disk", func(t *testing.T) {
+			vendor := "testVendor"
+			chartName := "testChart"
+			newCharts := []*ChartWrapper{
+				{
+					Chart: &chart.Chart{
+						Metadata: &chart.Metadata{
+							APIVersion: "v2",
+							Name:       chartName,
+							Version:    "2.3.4",
+						},
+					},
+				},
+				{
+					Chart: &chart.Chart{
+						Metadata: &chart.Metadata{
+							APIVersion: "v2",
+							Name:       chartName,
+							Version:    "1.2.3",
+						},
+					},
+				},
+			}
+			repoRoot := t.TempDir()
```

Comment:
I think it would be good to check that those charts are not already present on the disk before calling `writeCharts`